src/UI/Art/SpriteArtist/PlayerSetupAiArtist.java

```diff
@@ -0,0 +1,111 @@
+    {
+      // Figure out the longest-length name.
+      for( int i = 0; i < aiOptions.size(); ++i )
+      {
+        if( aiOptions.get(i).getName().length() > maxAiNameLength ) maxAiNameLength = aiOptions.get(i).getName().length();
+      }
+
+      // Make a string of all spaces of the required length so we can build nameplates of the same length. Soooo hacky, I know.
+      StringBuilder buf = new StringBuilder();
+      for( int n = 0; n < maxAiNameLength; ++n ) buf.append(" ");
+
+      // Build and store off our nameplates for posterity.
+      for( int i = 0; i < aiOptions.size(); ++i )
+      {
+        BufferedImage frame = SpriteUIUtils.makeTextFrame(SpriteUIUtils.MENUBGCOLOR, SpriteUIUtils.MENUFRAMECOLOR, buf.toString(), 3, 2);
+        SpriteLibrary.drawImageCenteredOnPoint(frame.getGraphics(), SpriteLibrary.getTextAsImage(aiOptions.get(i).getName()), frame.getWidth()/2, frame.getHeight()/2, 1);
+        aiNameplates.put(i, frame);
+      }
+    }
+
+    // Calculate the vertical space each name will consume.
+    int nameplateHeight = aiNameplates.get(0).getHeight();
+    int nameBuffer = nameplateHeight/2;
+    int nameplateYSpace = nameplateHeight+nameBuffer;
+
+    // Find where the zeroth option should be drawn.
+    int highlightedAi = control.getSelectedAiIndex();
+    int drawY = nameBuffer/2;
+
+    // Draw all of the visible AI names.
+    for(int i = 0; i < aiNameplates.size(); i++, drawY += (nameplateYSpace))
+    {
+      // Only bother to draw it if it is on-screen.
+      if( (drawY > -nameplateYSpace/2) && ( drawY < myHeight+(nameplateYSpace/2) ) )
+      {
+        BufferedImage aiNameplate = aiNameplates.get(i);
+
+        int drawX = nameBuffer;
+        myG.drawImage(aiNameplate, drawX, drawY, null);
+
+        // Draw the cursor if this panel is highlighted
+        if( highlightedAi == i )
+        {
+          SpriteUIUtils.drawCursor(myG, drawX, drawY, aiNameplate.getWidth(), aiNameplate.getHeight(), playerColor);
+        }
+      }
+    }
+
+    ///////////////// AI Info ///////////////////////
+    int aiNameplateZoneWidth = aiNameplates.get(0).getWidth() + nameplateHeight;
+    int infoBuffer = nameplateHeight/2;
+    int aiInfoZoneWidth = myWidth - aiNameplateZoneWidth;
+    int infoX = aiNameplateZoneWidth + infoBuffer;
+    int infoY = infoBuffer;
+    int infoW = aiInfoZoneWidth - infoBuffer * 2;
+    int infoH = myHeight - infoBuffer*2;
+    String text = "If I had anything helpful to say about this AI, I would write it here.\n" +
+        "I don't though, so all you get is this useless blurb.";
+
+    // Draw a nice frame and populate it with some useful info about the AI.
+    myG.setColor(new Color(255, 255, 255, 80));
+    myG.fillRect(infoX, infoY, infoW, infoH);
+    myG.setColor(new Color(100, 115, 130));
+    myG.drawRect(infoX, infoY, infoW, infoH);
+
+    // Draw the composed image to the window at scale.
+    g.drawImage(image, 0, 0, dimensions.width, dimensions.height, null);
```

Comment:
Why do you specify width and height? They should be equal to your current dimensions anyway.

Reply:
Because `image` is at pixel-space size, and `dimensions` is at screen-space size. If I take that out it won't scale the image to match the screen.